### PR TITLE
925 datasets auto approval

### PR DESCRIPTION
**Pull request by SofiaSazonova**
### Feature or Bugfix

- Feature

### Detail
- AutoApprove property was added to UI and API
- If it is True, then the sharing request is approved right after submission

### Relates
https://github.com/data-dot-all/dataall/issues/925

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)? Yes
  - Is the input sanitized? No
  - What precautions are you taking before deserializing the data you consume? Default deserialiser is used
  - Is injection prevented by parametrizing queries? YES
  - Have you ensured no `eval` or similar functions are used? N/A
- Does this PR introduce any functionality or component that requires authorization? No
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features? No
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users? No
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: backend/dataall/modules/dataset_sharing/services/share_object_service.py [hunk 1]
```diff
@@ -236,7 +247,7 @@ def approve_share_object(cls, uri: str):
             session.add(approve_share_task)
 
         Worker.queue(engine=context.db_engine, task_ids=[approve_share_task.taskUri])
-
+        print('Hurray!')
```
Comment (noah-paige) [inline backend/dataall/modules/dataset_sharing/services/share_object_service.py:250]: Great print statement :) ... but to remove haha

Comment (SofiaSazonova) [reply backend/dataall/modules/dataset_sharing/services/share_object_service.py]: Oops. Removed)

**Comment by noah-paige** [comment]
One thought - do we want to expose the value of `autoApprove` flag somewhere on the dataset overview

**Comment by noah-paige** [comment]
> > One thought - do we want to expose the value of `autoApprove` flag somewhere on the dataset overview
> 
> I thought about it. From my POV auto-approval is close to confidentiality. But it seems that confidentiality is not exposed anywhere in views. Happy to discuss


I think we do expose `confidentiality` as `Classification` in the Dataset Overview Tab under the `Governance & Classification` Card

I think we can also expose `autoApprove` values for the dataset in the same Card

**Comment by noah-paige** [comment]
Tested in AWS:
- [x] CICD Pipeline Succeeds

With an existing dataset:
- [x] Create Share Request
- [x] Approve Share Request
- [x] Revoke Share and Update Dataset to AutoApprove True
- [x] Create Share and Auto Approved

With a new dataset:
- [x] Create Dataset with AutoApprove False
- [ ] Create Share Request
- [ ] Approve Share Request

With an imported dataset
- [x] Create Dataset with AutoApprove True
- [ ] Create Share Request and Auto Approved

**Comment by SofiaSazonova** [comment]
> > > One thought - do we want to expose the value of `autoApprove` flag somewhere on the dataset overview
> > 
> > 
> > I thought about it. From my POV auto-approval is close to confidentiality. But it seems that confidentiality is not exposed anywhere in views. Happy to discuss
> 
> I think we do expose `confidentiality` as `Classification` in the Dataset Overview Tab under the `Governance & Classification` Card
> 
> I think we can also expose `autoApprove` values for the dataset in the same Card
> 
> UPDATE: Finished testing and everything working as expected! I am inclined to expose `autoApprove` value in `dataall/frontend/src/modules/Datasets/components/DatasetGovernance.js` but open to other ides if people think differently - from there can approve and merge

Done

**Comment by noah-paige** [comment]
Verified - showing Auto-Approval flagas part of Dataset Overview now

**Review by noah-paige — APPROVED** [review]
Looks good! Approving

**Pull request merged**